fix(ticks): Ensure model value is current when custom translate funct…

trodi opened the pull request:
…ion runs for tick values

Ensure translate function behaves the same for keyboard commands and drag events as mouse clicks by ensuring the model is applied before updateHandles is called. Currently when using mouse clicks to change the model of the slider, the translate function is called after the model is updated. My change ensures this same state when you drag a handle or use keyboard shortcuts. I've parameterized `applyModel` so that we only call the onchange event once if a change to the model has happened. @ValentinH, if there is a preferred approach to this other than I what I did, please advise and I'll update my PR.

I've made a second commit with the changes from running a build so the following fiddle will work. I can destroy once you've seen it working before anything gets merged. Updated [JSFiddle](http://jsfiddle.net/Lomg3b5z/2/) from issue to show now working example.

#498

ValentinH commented:
Thanks for this PR, it looks pretty clean! Could you had a test case for the new parameter you added to the  `applyModel` method?

trodi commented:
* I've added a test for `applyModel`
* I've removed the dummy commit of the example build changes
* I've squashed
* As I was writing the test checking whether `callOnChange` gets called, I noticed that that other tests that check the same thing in different circumstances don't test that it is only called once. I added a second commit updating all the tests so they explicitly expect the function called the correct number of times. This should provide regression if someone later updates/refactors the code around this.
  * I can squash if you like this all together **or** I can create a new PR for this test _refactoring_. Please advise.

ValentinH commented:
Thanks, best PR ever! :)

ValentinH commented:
Released under 6.0.1
